massa-api/src/proxy.rs

```diff
@@ -0,0 +1,212 @@
+//! Middleware that proxies requests at a specified URI to internal
+//! RPC method calls.
+
+use hyper::header::{ACCEPT, CONTENT_TYPE};
+use hyper::http::HeaderValue;
+use hyper::{Body, Method, Request, Response, StatusCode, Uri};
+use jsonrpsee::core::error::Error as RpcError;
+use std::error::Error;
+use tower::{Layer, Service};
+// use std::error::Error;
+use jsonrpsee::core::params::ArrayParams;
+use jsonrpsee::core::traits::ToRpcParams;
+use jsonrpsee::types::{Id, RequestSer};
+use serde_json::Value;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use urlencoding::decode;
+
+pub const URI_ADDRESSES: &str = "/addresses";
+pub const URI_BLOCKS: &str = "/blocks";
+pub const URI_ENDORSEMENTS: &str = "/endorsements";
+pub const URI_OPERATIONS: &str = "/operations";
+
+/// Layer that applies [`MassaProxyGetRequest`] which proxies the `GET /path` requests to
+/// specific RPC method calls and that strips the response.
+///
+/// See [`MassaProxyGetRequest`] for more details.
+#[derive(Debug, Clone)]
+pub struct MassaProxyGetRequestLayer {}
+
+impl MassaProxyGetRequestLayer {
+    /// Creates a new [`ProxyGetRequestLayer`].
+    ///
+    /// See [`MassaProxyGetRequest`] for more details.
+    pub fn new() -> Result<Self, RpcError> {
+        Ok(Self {})
```

Comment:
Why this the `new`is empty ? may be we could add some checks here and throw an error(failfast)